src/check_config_provider/redis_config_provider.rs

```diff
@@ -153,39 +149,35 @@ impl RedisConfigProvider {
                 .set(config_payloads.len() as f64);
 
             for config_payload in config_payloads {
-                let config: CheckConfig = rmp_serde::from_slice(&config_payload)
-                    .map_err(|err| {
-                        tracing::error!(?err, "config_consumer.invalid_config_message");
-                    })
-                    .unwrap();
-                manager
-                    .get_service(partition.number)
-                    .get_config_store()
-                    .write()
-                    .unwrap()
-                    .add_config(Arc::new(config));
+                let config: Result<CheckConfig, _> = rmp_serde::from_slice(&config_payload);
+
+                match config {
+                    Ok(config) => {
+                        manager
+                            .get_service(partition.number)
+                            .get_config_store()
+                            .write()
+                            .expect("Lock should not be poisoned")
+                            .add_config(Arc::new(config));
+                    }
+                    Err(err) => tracing::error!(?err, "config_consumer.invalid_config_message"),
+                }
             }
-            let partition_loading_time = (Utc::now() - partition_start_loading)
-                .to_std()
-                .unwrap()
-                .as_secs_f64();
             metrics::histogram!(
                 "config_provider.initial_load.partition.duration",
                 "histogram" => "timer",
                 "uptime_region" => region.clone(),
                 "partition" => partition.number.to_string(),
             )
-            .record(partition_loading_time);
+            .record(partition_start_loading.elapsed().as_secs_f64());
```

Comment:
minor nit, I sort of prefer to keep the variable name here since it helps document what this value is, vs having to parse this expression and understand what it is